### DISCUSSION TITLE
hb_feature 'liga' dosen't work sometime.

### DISCUSSION THREAD

**Discussion by totake0224** [harfbuzz/harfbuzz#4224] [Q&A]
I'm sorry I could not explain my situation at title.
I'm using harfbuzz from php extention.
I made two functions.
first function is to get alternatives and glyph id from font.
second function is to get path from glyph id.
I first set hb_feature 'liga' at second function only .but it doesn't work.
but I set hb_feature 'liga' at both function. it works.
first function's hb_feature 'liga'  1 or 0 doesn't affect second function result.
for example
In first function
```c
hb_shape(hb_font, buffer,NULL, 0);
````
In second function
```c
userfeatures[0].tag = HB_TAG('l','i','g','a');
userfeatures[0].value = 1;
userfeatures[0].start = HB_FEATURE_GLOBAL_START;
userfeatures[0].end = HB_FEATURE_GLOBAL_END;
hb_shape(hb_font, buffer, userfeatures, 1);
````
second function return is not reflect liga.
but
In first function
```c
userfeatures[0].tag = HB_TAG('l','i','g','a');
userfeatures[0].value = 0;
userfeatures[0].start = HB_FEATURE_GLOBAL_START;
userfeatures[0].end = HB_FEATURE_GLOBAL_END;
hb_shape(hb_font, buffer, userfeatures, 1);
````
In second function
```c
userfeatures[0].tag = HB_TAG('l','i','g','a');
userfeatures[0].value = 1;
userfeatures[0].start = HB_FEATURE_GLOBAL_START;
userfeatures[0].end = HB_FEATURE_GLOBAL_END;
hb_shape(hb_font, buffer, userfeatures, 1);
````
second function return reflect 'liga=1'

both function called separately from php.
I got desired result by calling hb_shape with userfeatures at both functions.
but I have strange feeling at this result.
Please point out if there is something wrong at my code.

Thnak you.

**Comment by behdad** [comment]
What font are you using? Can you attach images?
